pyproject.toml

```diff
@@ -1,20 +1,87 @@
+[project]
+name = "nipy"
+dynamic = ['version']
+license = {file = "LICENSE"}
+description  = 'A python package for analysis of neuroimaging data'
+readme = 'README.rst'
+classifiers = ["Development Status :: 3 - Alpha",
+               "Environment :: Console",
+               "Intended Audience :: Science/Research",
+               "License :: OSI Approved :: BSD License",
+               "Operating System :: OS Independent",
+               "Programming Language :: Python",
+               "Topic :: Scientific/Engineering"]
+dependencies = [
+    'numpy>=1.22',
+    'scipy>=1.8',
+    'nibabel>=3.2',
+    'sympy>=1.9',
+    'transforms3d'
+]
+
+[[project.maintainers]]
+name = 'nipy developers'
+email = 'neuroimaging@python.org'
+
+[project.urls]
+homepage = 'https://nipy.org/nipy'
+documentation = 'http://nipy.org/nipy/documentation.html'
+source = 'https://github.com/nipy/nipy'
+download = 'https://pypi.org/project/nipy/#files'
+tracker = 'https://github.com/nipy/nipy/issues'
+
+[project.optional-dependencies]
+build = [
+    'meson-python>=0.13',
+    'wheel',
+    'setuptools>=67',
+    'ninja',
+    'Cython>=3',
+    'numpy>=1.22',
+    'spin==0.3',
+    'build',
+]
+default = [
+    'numpy>=1.22',
+    'scipy>=1.8',
+    'nibabel>=3.2',
+    'sympy>=1.9',
+    'transforms3d'
+]
+developer = [
+    'pre-commit',
+    'rtoml',
+]
+docs = [
+    'sphinx>=7.0',
+    'numpydoc>=1.6.0',
+    'matplotlib',
+    'texext',
+    'ipython'
+]
+optional = [
+    'matplotlib>=3',
+]
+test = [
+    'matplotlib>=3',
+    'pytest>=7.2',
+    'pytest-cov>=4.0',
+    'pytest-doctestplus'
+]
+
 [build-system]
 build-backend = "mesonpy"
 requires = [
   "meson-python>=0.13",
+  "ninja",
```

Comment:
I'm able to build without this. Is `ninja` a dependency of `meson-python` that you just want to be explicit about? Do you actually depend on ninja-specifics, or could the other build tools stop using it without you needing to change anything?

Reply:
I was getting errors trying to build without it, for the wheel builds.   I'm afraid I don't understand the build mechanics well enough to respond.   If ninja is missing, with meson-python use something else?

Reply:
How are you building? I'm using `python -m build`.

Reply:
Hmm. Possibly an old pip? I just created a fresh environment:

```
$ mamba create -n tmp python=3.11
$ mamba activate tmp
$ pip wheel .
Processing /home/chris/Projects/nipy/nipy
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: nipy
  Building wheel for nipy (pyproject.toml) ... done
  Created wheel for nipy: filename=nipy-0.5.1.dev1-cp311-cp311-linux_x86_64.whl size=1916097 sha256=7ff5b2a9a9fe053c913dd2e487ee8c8fd11991893cf96e0f806846c79d17cd8b
  Stored in directory: /tmp/pip-ephem-wheel-cache-sn4mrur1/wheels/a1/58/f7/1af5aca9b8327defaf19028fc61c6f4850bd54b33e92f89a43
Successfully built nipy
$ pip --version
pip 23.3.1 from /home/chris/mambaforge/envs/tmp/lib/python3.11/site-packages/pip (python 3.11)
```